openupgrade_scripts/scripts/hr_holidays/17.0.1.6/pre-migration.py

```diff
@@ -0,0 +1,97 @@
+# Copyright 2024- Le Filament (https://le-filament.com)
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl).
+from openupgradelib import openupgrade
+
+_model_renames = [
+    ("hr.leave.stress.day", "hr.leave.mandatory.day"),
+]
+
+_table_renames = [
+    ("hr_leave_stress_day", "hr_leave_mandatory_day"),
+]
+
+
+def _map_leave_accrual_level_action(cr):
+    openupgrade.logged_query(
+        cr,
+        """
+        UPDATE hr_leave_accrual_level
+        SET action_with_unused_accruals = 'all'
+        WHERE action_with_unused_accruals = 'postponed';
```

Comment:
I think setting this field to `all` always is incorrect; it must depend on the `maximum_leave` field.
Otherwise, it can never be applied according to [this code](https://github.com/odoo/odoo/blob/7bf18616e4ae39c5e712339cfd560b57018a93de/addons/hr_holidays/models/hr_leave_allocation.py#L521-L524
)

V16
![image](https://github.com/user-attachments/assets/d87bd494-bdfa-495f-894c-b6175a9d45d4)
V17
![image](https://github.com/user-attachments/assets/8ae4d317-cd22-4986-b394-436d6819db66)

V17 expected behavior
![image](https://github.com/user-attachments/assets/39430ff9-6f42-47fc-bae1-a6213eb5fc44)

```suggestion
        UPDATE hr_leave_accrual_level
        SET action_with_unused_accruals = CASE 
                WHEN maximum_leave > 0 THEN 'maximum' 
                ELSE 'all' 
                END
        WHERE action_with_unused_accruals = 'postponed';
```